DocsScraper/src/DocsScraper.jl

```diff
@@ -0,0 +1,235 @@
+
+# Arguments
+- `node`: The root HTML node 
+- `args[1]`: heading_hierarchy - Dict used to store metadata
+- `args[2]`: parsed_text - Vector of Dicts to store parsed text and metadata
+- `args[3]`: child_new - Bool to specify if the current block (child) is part of previous block or not. 
+                        If it's not, then a new insertion needs to be created in parsed_text
+- `args[4]`: prev_text - String which contains previous text
+"""
+function process_node!(node::Gumbo.HTMLElement, args...)
+    heading_hierarchy = length(args) >= 1 ? args[1] : Dict{String,Any}()
+    parsed_text = length(args) >= 2 ? args[2] : Vector{Dict{String,Any}}()
+    child_new = length(args) >= 3 ? args[3] : true
+    prev_text = length(args) >= 4 ? args[4] : ""
+
+    tag_name = String(Gumbo.tag(node))
+
+    ## return statements
+    # `is_code` specifies if the block is code or not
+    is_code = false
+
+    # `is_text_inserted` is a flag which specifies if the text is inserted in the subsequent recursion iteration. 
+    is_text_inserted = false
+
+    # process headings. Working: If the current tag is a heading, then, we remove all the smaller headings we stored (for metadata) from  
+    # heading_hierarchy dictionary and add the current heading tag
+    # This is done because, in the heading hierarchy, if we come back to the current heading, 
+    # then there is no need to store the smaller headings we encountered earlier
+    if startswith(tag_name, "h") && isdigit(last(tag_name))
+
+        # Clear headings of equal or lower level
+        for k in collect(keys(heading_hierarchy))
+            if Base.parse(Int, last(k)) >= Base.parse(Int, last(tag_name))
+                delete!(heading_hierarchy, k)
+            end
+        end
+        heading_hierarchy[tag_name] = strip(Gumbo.text(node))
+        push!(parsed_text,
+            Dict("heading" => strip(Gumbo.text(node)),
+                "metadata" => copy(heading_hierarchy)))
+        return "", is_code, is_text_inserted
+
+        # if the current tag is <code>, then get it's textual value and return it with backticks (``)
+    elseif tag_name == "code"
+        # Start a new code block
+        code_content = "`" * strip(Gumbo.text(node)) * "`"
+        is_code = true
+        return code_content, is_code, is_text_inserted
+    else
+        text_to_insert = ""
+
+        # Recursively process the child node for text content
+        children = collect(AbstractTrees.children(node))
+        num_children = length(children)
+
+        for (index, child) in enumerate(children)
+
+            # if the current tag belongs in the list, it is assumed that all the text/code should be part of a single paragraph/block, unless,
+            # there occurs a code block with >50 chars, then, previously parsed text is inserted first, then the code block is inserted. 
+            if tag_name in ["p", "li", "dt", "dd", "pre", "b", "strong", "i", "cite", "address", "em", "td"]
```

Comment:
Could we simplify the code backtick handling by focusing on the PARENT (`parent(node)`) attribute once we get to the `code` block?

The simple rule I see is to decide the code separator when we get the code block based on its parent, eg,
```
function code_fences(node::HTMLElement)
  
  if parent(tag(node)) == :pre
    # implicitly assumes julia, can be easily improved by checking the code CLASS attribute
    "```julia", "```"
  else
    "`", "`"
  end
end

open_fence,close_fence = code_fences(node)
# mock implementation
chunk = open_fence * content * close_fence
```




I'm referring to: https://github.com/JuliaWeb/Gumbo.jl/blob/afc2b2b83501d483e416d86063d98d567968fea7/src/htmltypes.jl#L15
To make your life easy, export it on module level `using AbstractTrees: parent, children` (to have easily available).